modules/programs/neovim.nix

```diff
@@ -217,6 +217,38 @@ in {
           This option is mutually exclusive with <varname>configure</varname>.
         '';
       };
+
+      coc = {
+        enable = mkEnableOption "Coc";
+
+        extraConfig = mkOption {
```

Comment:
Why not use a `settings` option, using `formats.json`?

Reply:
Yes, `extraConfig` will not be accepted if the configuration can use structured types.

Reply:
As an example: https://github.com/nix-community/home-manager/blob/db00b39a9abec04245486a01b236b8d9734c9ad0/modules/services/hound.nix.

This module uses `pkgs.formats.json`.